lib/auth/tls_test.go

```diff
@@ -3382,6 +3382,98 @@ func TestEventsPermissions(t *testing.T) {
 	}
 }
 
+// TestEventsPermissionsPartialSuccess verifies that in partial success mode NewWatcher can still succeed
+// if caller lacks permission to watch only some of the requested resource kinds.
+func TestEventsPermissionsPartialSuccess(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		name                   string
+		watch                  types.Watch
+		identity               TestIdentity
+		expectedConfirmedKinds []types.WatchKind
+	}{
+		{
+			name: "no permission for any of the requested kinds",
+			watch: types.Watch{
+				Kinds: []types.WatchKind{
+					{Kind: types.KindUser},
+					{Kind: types.KindRole},
+				},
+				AllowPartialSuccess: true,
+			},
+			identity:               TestBuiltin(types.RoleDiscovery),
```

Comment:
Yeah, I took this shortcut because I hadn't realized how easy it was to define a custom role for the test. Fixed in https://github.com/gravitational/teleport/pull/24080/commits/110add42183b83f38736b58c12b5d041e40eddf4